client/lib/route/path.ts

```diff
@@ -25,6 +25,11 @@ export function getSiteFragment( path: URLString ): SiteSlug | SiteId | false {
 		return false;
 	}
 
+	// Avoid confusing the subscription ID for the site ID in gifting checkouts.
+	if ( basePath.includes( '/gift/' ) && basePath.includes( '/checkout/' ) ) {
```

Comment:
If `/checkout`/ is always in the beginning, does it make sense to make this a bit stricter with that in mind?

```suggestion
	if ( basePath.includes( '/gift/' ) && basePath.startsWith( '/checkout/' ) ) {
```

Reply:
Sounds good @tyxla, I will commit this.